pkg/v1/cli/command/core/doc.go

```diff
@@ -10,46 +10,62 @@ import (
 	"os"
 	"text/template"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/spf13/cobra/doc"
 
 	cliv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cli/v1alpha1"
 	cli "github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli"
 	coreTemplates "github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/core/templates"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/pluginmanager"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/config"
 )
 
 // DefaultDocsDir is the base docs directory
 const DefaultDocsDir = "docs/cli/commands"
+const ErrorDocsOutputFolderNotExists = "error reading docs output folder '%v', make sure folder exists or provide docs output folder as input value to '--docs-dir' argument"
 
 var (
 	docsDir string
 )
 
 func init() {
-	genAllDocsCmd.Flags().StringVarP(&docsDir, "docs-dir", "d", DefaultDocsDir, "destination for docss output")
+	genAllDocsCmd.Flags().StringVarP(&docsDir, "docs-dir", "d", DefaultDocsDir, "destination for docs output")
 }
 
 var genAllDocsCmd = &cobra.Command{
 	Use:    "generate-all-docs",
 	Short:  "Generate Cobra CLI docs for all plugins installed",
 	Hidden: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
+
+		if docsDir == "" {
+			docsDir = DefaultDocsDir
+		}
+		if _, err := os.Stat(docsDir); err != nil {
+			return errors.Wrap(err, fmt.Sprintf(ErrorDocsOutputFolderNotExists, docsDir))
+		}
 		// Generate standard tanzu.md command file
 		if err := genCoreCMD(cmd); err != nil {
 			return fmt.Errorf("error generate core tanzu cmd markdown %q", err)
 		}
 
-		plugins, err := cli.ListPlugins()
+		var pluginDescriptions []*cliv1alpha1.PluginDescriptor
+		var err error
+		if config.IsFeatureActivated(config.FeatureContextAwareCLIForPlugins) {
+			pluginDescriptions, err = pluginmanager.InstalledPluginsDescriptors()
+		} else {
+			pluginDescriptions, err = cli.ListPlugins()
```

Comment:
Should we continue using the deprecated function, or should we use the recommended replacement?

Reply:
there are some git integration tests still depend on deprecated code, I have created new issue to address it and remove the deprecated code - https://github.com/vmware-tanzu/tanzu-framework/issues/3224

Reply:
> there are some git integration tests still depend on deprecated code, I have created new issue to address it and remove the deprecated code - #3224

Sorry, I did't express myself well.  We can keep the deprecated function `ListPlugins`, but in line 58 above, do we still need to use it or can we use the replacement function?

Reply:
We still need to use it, initially I have removed it, later Anuj suggested better to keep it until we remove the deprecated code completely.